### PR TITLE
More options for displaying sticky post in Posts block

### LINKED ISSUE #1325: Posts block doesn't show the Sticky post as a featured post.
### Description:
A user has asked about this on the wp.org forum that if he can use the sticky post as a featured post in Otter's posts block.

When you use the Posts block and turn on the featured post option, then the sticky post doesn't appear as a featured post. The sticky post appears at the top of the Blog/Archive page so his query is if it can be also used here and if it could appear as a featured post.

P.s  - I am not sure If I should add an enhancement tag or the bug tag.😅

### How to reproduce:

1. Create some demo posts.
2. Go to any random post and make it a sticky post by going to post settings [See attached screenshot in reference].
3. Install Otter and Create a new page.
4. Use Otter's **Posts Block** and turn on the featured post option. 
5. Go to the page and observe that the post which is featured on top is the latest post that was created by you. If you will create a new post then the new post will appear as a Featured post.
6. Now go to the Blog/Archive page and see that the sticky post appears at the top instead of the latest post.

### Expected behavior:
The posts block should show the Sticky post as a featured post. 

### Current behavior:
The posts block doesn't show the Sticky post as a featured post. 

### Reference:

image
image

image
image

The user asked here : https://wordpress.org/support/topic/can-i-keep-the-sticky-post-as-the-featured-post-in-the-posts-block/
### Technical info
* WordPress version: 6.1
* Plugin version: 2.1.3

### PR DESCRIPTION
**Pull request by Soare-Robert-Daniel**
Closes #1325 .

### Summary

Add the option to display sticky posts on first positions when Featured Post is enabled.

### Screenshots 

image

----

### Test instructions

1. Make some posts and publish them. Mark some of them as sticky posts.
2. Create another post with our Posts block.
3. Activate Featured Post and play with the option.

---- 

### Checklist before the final review

- [ ] Visual elements are not affected by independent changes.
- [ ] It is at least compatible with the [minimum WordPress version](https://wordpress.org/plugins/otter-blocks/).
- [ ] It loads additional script in frontend only if it is required.
- [ ] Does not impact the [Core Web Vitals](https://web.dev/vitals/).
- [ ] In case of deprecation, old blocks are safely migrated.
- [ ] It is usable in Widgets and FSE.
- [ ] Copy/Paste is working if the attributes are modified.

### CODE REVIEW

**Comment by irinelenache** [comment]
@Soare-Robert-Daniel Tested the changings and here are my findings:

- [ ] The featured post is different in editor and frontend, in editor it will always be the last post instead of the sticky one (if it exists)
editor: https://vertis.d.pr/i/8H4wY4
frontend: https://vertis.d.pr/i/iyE3nw

- [ ] If the posts block is set to show the latest X items and the sticky post is older and doesn't make the X items list, it will not appear in the block
all the posts in the instance, with an older one being sticky: https://vertis.d.pr/i/x67Gvz
the block in frontend, with 4 items selected: https://vertis.d.pr/i/kXGtIz

Besides these, everything seems to work fine 👍

**Comment by Soare-Robert-Daniel** [comment]
> * [ ]  If the posts block is set to show the latest X items and the sticky post is older and doesn't make the X items list, it will not appear in the block
>   all the posts in the instance, with an older one being sticky: https://vertis.d.pr/i/x67Gvz
>   the block in frontend, with 4 items selected: https://vertis.d.pr/i/kXGtIz

Can you offer me the instance?

**Comment by Soare-Robert-Daniel** [comment]
> * [ ]  If the posts block is set to show the latest X items and the sticky post is older and doesn't make the X items list, it will not appear in the block
>   all the posts in the instance, with an older one being sticky: https://vertis.d.pr/i/x67Gvz
>   the block in frontend, with 4 items selected: https://vertis.d.pr/i/kXGtIz

I think this is happening because the post has the same as the other 4, and the retrieving might not respect the ordering from the interface.

@HardeepAsrani I am right on this?

**Comment by Soare-Robert-Daniel** [comment]
> @Soare-Robert-Daniel I'm not sure, I suppose Sticky should always appear but you can check with the Core behavior to see how they handle it. Instead of this new select option, we can just have a checkbox for "Include Sticky Posts."

Is not about including, but a priority. When you select Sticky, the sticky post will have priority. The issue mentioned by Irinel there might be a problem with his post configuration since many of having the same date, I think that WP did not respect the ordering since it give him the first post with the same date.

image

**Comment by HardeepAsrani** [comment]
@Soare-Robert-Daniel Probably this: https://developer.wordpress.org/reference/functions/get_posts/#description

**Comment by Soare-Robert-Daniel** [comment]
After testing in the instance. Nothing to be done. That is just how the WP gives the result on the API that we use on both BACKEND AND FRONTEND. 

Our block:
image

After testing with Query Loop, I found a significant inconsistency.

In the Backend/Gutenberg, it shows the same as ours.

image

But in Frontend is using the correct ordering.
image

The usage of `wp_get_recent_posts` is consistent with `select( 'core' ).getEntityRecords( 'postType', 'post', latestPostsQuery )` for us.

But it seems Query Loop is using different things, and only the one in the frontend is correct.

**Comment by HardeepAsrani** [comment]
@Soare-Robert-Daniel I'm guessing in front it uses WP_Query that can handle that: https://developer.wordpress.org/reference/classes/wp_query/

**Comment by HardeepAsrani** [comment]
@Soare-Robert-Daniel All good here?

**Comment by Soare-Robert-Daniel** [comment]
@HardeepAsrani, no. I need to see how Query Loop queries get the right order for that edge case highlighted by Irinel.

**Comment by Soare-Robert-Daniel** [comment]
After more investigation, I found the Post listing in the interface is sorted in ascending mode for post that have the same publishing date. 

In Irinel's picture,  the sticky post (`Appearance Guide`) has lower ID that `Improve him..`.

Our block is doing the correct in picking `Improve him..` since it is the newest. But Query Loop does not respect this, the posts with the date are in ASC order, which should not have happened. The result from Editor are different from the frontend, which make Query Loop to be misleading . 

image

That thing is more of an issue with the Core.

File: inc/render/class-posts-grid-block.php [hunk 1]
```diff
@@ -7,6 +7,9 @@
 
 namespace ThemeIsle\GutenbergBlocks\Render;
 
+use WP_Post;
+use WP_Query;
```
Comment (HardeepAsrani) [inline inc/render/class-posts-grid-block.php:11]: You can remove the used class.

Comment (HardeepAsrani) [reply inc/render/class-posts-grid-block.php:11]: Apart from that, seems to be good and work fine, you can merge it.

**Pull request merged**